includes/User.php

```diff
@@ -2167,6 +2167,8 @@ public function setPassword( $str ) {
 				}
 				throw new PasswordError( wfMsgExt( $message, array( 'parsemag' ), $params ) );
 			}
+	            	//Delete all access tokens
+	            	$this->removeAllTokens();
```

Comment:
I prefer hooks over adding stuff to MW core (despite the fact that we no longer update it).